docs/source/libmav/libmav.py

```diff
@@ -0,0 +1,1041 @@
+        
+        The method blocks and waits for the expected message to be received.
+        It either returns the message or exits after a timeout.
+        
+        ```python
+        # create a message
+        messageToSend = message_set.create('SOME_SEND_MESSAGE')
+
+        // Start watching for the response before sending
+        expectation = connection.expect("EXPECTED_RESPONSE_MESSAGE");
+
+        # send the message
+        connection.send(messageToSend);
+
+        # receive the expected message, with a 1s timeout
+        messageReceived = connection.receive(expectation, 1000);
+        ```
+
+        > **Note**
+        > - The synchonous recieve methods wait on only one message, and will ignore any other messages that arrive in the meantime. 
+        >   If you want to wait on more than one message you will need to use `add_message_callback()`.
+        > - Use the `receive(self, messageName, timeout)` version to wait on a message that is being streamed..
+
+        Args: 
+            expectation (ExpectationWrapper): An expectation, created using `Connection.expect()`.
+            timeout (int): Timeout in miliseconds. By default there is no timeout.
+
+        Returns:
+            Message: The message that was waited on.
+        """   
+
+    def remove_message_callback(self, callback_handle):
+        """Removes a message callback that was previously added using `add_message_callback()`.
+        
+        The code below shows how you might add and then remove a callback.
+        
+        ```python   
+        callback_handle = connection.add_message_callback(lambda msg: print(f'{msg.name}'))
+        connection.remove_message_callback(callback_handle)
+        ```
+
+        Args:
+            callback_handle (callback): The handle of the callback to remove.
+        """
+
+    def send(self, message):
+        """Send a message.
+
+        The code below shows how you might create a message and then send it.
+        See `MessageSet` and `Message` for more information on how you set the message properties.
+
+        ```python
+        # create a message
+        heartbeat_message = message_set.create('HEARTBEAT')
+        
+        # send the message
+        conection.send(heartbeat_message);
+        ```
+
+        Args: 
+            message (Message): The message to send.
+        """
+        pass 
+
+class ConnectionPartner():
+    """A class representing the remote partner of a `Connection`.
+    
+    Objects of this type are returned by `Connection.partner()`.
+    Libmav applications can use these to differentiate between connections on the same `NetworkRuntime` interface.
+    This is needed for server interfaces (`UDPServer` or `TCPServer`), where more than one external system might connect to the port libmav is listening on,        
+
+    <!-- Serial connection just set is_uart because you don't need to know what the address is for differentiation? -->
+    -->
```

Comment:
```suggestion
For serial connections `is_uart` is set (there is no need to differentiate partners, because there can only be one partner on the interface).
```